src/main/main.ts

```diff
@@ -917,6 +917,24 @@ const createWindow = async () => {
         mainWindow.webContents.openDevTools();
     }
 
+    let hasUnsavedChanges = false;
+    ipcMain.on('update-has-unsaved-changes', (_, value) => {
+        hasUnsavedChanges = value;
+    });
+
+    mainWindow.on('close', (event) => {
+        if (hasUnsavedChanges) {
+            const choice = dialog.showMessageBoxSync(mainWindow, {
+                type: 'question',
+                buttons: ['Yes', 'No'],
```

Comment:
By default, the first button (Yes) will be selected (focused). However, we probably want our users to save those changes.
```suggestion
                buttons: ['Yes', 'No'],
                defaultId: 1,
```